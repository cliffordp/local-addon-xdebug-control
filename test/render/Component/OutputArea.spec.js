const React = require( 'react' )
const Component = require( 'react' ).Component
const renderer = require( 'react-test-renderer' )

const context = {
	'React': React,
}

const OutputArea = require( './../../../src/Component/OutputArea' )( context )

describe( '<OutputArea/>', function () {
	const defaultProps = {
		children: ['foo', 'bar', 'baz'].map( ( e ) => {
			return (
				<p>{e}</p>
			)
		} ),
		centerX: true,
		centerY: true,
	}

	it( 'renders correctly with props', function () {
		const component = renderer.create( <OutputArea {...defaultProps}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly without children', function () {
		const props = Object.assign( {}, defaultProps, {children: []} )
		const component = renderer.create( <OutputArea {...defaultProps}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )
} )
