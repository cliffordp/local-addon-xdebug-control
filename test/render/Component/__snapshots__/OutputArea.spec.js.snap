// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutputArea/> renders correctly with props 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <p>
      foo
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <p>
      bar
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <p>
      baz
    </p>
  </div>
</div>
`;

exports[`<OutputArea/> renders correctly without children 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <p>
      foo
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <p>
      bar
    </p>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <p>
      baz
    </p>
  </div>
</div>
`;
